subiquitycore/file_util.py

```diff
@@ -1,32 +1,67 @@
+# Copyright 2018-2022 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import datetime
+import grp
 import os
 import stat
 import tempfile
 
-_DEF_PERMS = 0o644
+_DEF_PERMS = 0o640
+_DEF_GROUP = 'adm'
 
 
-def write_file(filename, content, mode=None, omode="wb", copy_mode=False):
-    """Atomically write filename.
-    open filename in mode 'omode', write content, chmod to 'mode'.
-    """
-    if mode is None:
-        mode = _DEF_PERMS
+@contextlib.contextmanager
+def open_perms(filename, *, cmode=None, omode='w', copy_mode=False):
```

Comment:
No caller passes omode or copy_mode that I can see. Can we imagine a situation where we might use them?

Reply:
omode used to be used but disappears in this branch.  copy_mode was never used.  I'll remove both.